homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,639 @@
+"""Switches for AVM Fritz!Box functions."""
+from __future__ import annotations
+
+from collections import OrderedDict
+from functools import partial
+import logging
+from typing import Any
+
+from fritzconnection.core.exceptions import (
+    FritzActionError,
+    FritzActionFailedError,
+    FritzConnectionException,
+    FritzSecurityError,
+    FritzServiceError,
+)
+import xmltodict
+
+from homeassistant.components.network.util import async_get_source_ip
```

Comment:
This needs to be exposed in `homeassistant.components.network` before it can be used as it could get relocated later

Reply:
So should I stick on https://github.com/home-assistant/core/blob/5c5e43afc127fe420f314c01ee2e78d347631adc/homeassistant/util/__init__.py#L134
in the mean time ?

Reply:
In a new PR, something like:

```
diff --git a/homeassistant/components/network/__init__.py b/homeassistant/components/network/__init__.py
index 3f19103aca..1edf428078 100644
--- a/homeassistant/components/network/__init__.py
+++ b/homeassistant/components/network/__init__.py
@@ -19,6 +19,7 @@ from .const import (
 )
 from .models import Adapter
 from .network import Network
+from .util import async_get_source_ip
 
 ZEROCONF_DOMAIN = "zeroconf"  # cannot import from zeroconf due to circular dep
 _LOGGER = logging.getLogger(__name__)
diff --git a/tests/components/network/test_init.py b/tests/components/network/test_init.py
index 41d87d5a80..0a065693aa 100644
--- a/tests/components/network/test_init.py
+++ b/tests/components/network/test_init.py
@@ -444,3 +444,10 @@ async def test_interfaces_configured_from_storage_websocket_update(
             "name": "vtun0",
         },
     ]
+
+
+
+
+async def test_async_get_source_ip(hass):
+    """Test fetching the source ip."""
+    ...
\ No newline at end of file
```

Reply:
`get_local_ip` is probably fine in the mean time

Reply:
Done